app/src/androidTest/java/org/kiwix/kiwixmobile/main/TopLevelDestinationTest.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.main
+
+import androidx.test.rule.ActivityTestRule
+import org.junit.Rule
+import org.junit.Test
+import org.kiwix.kiwixmobile.utils.StandardActions.openDrawer
+
+class TopLevelDestinationTest {
+  @Rule
+  @JvmField
+  var activityTestRule = ActivityTestRule(KiwixMainActivity::class.java)
+
+  @Test
+  fun testTopLevelDestination() {
+    kiwixMain {
+      clickReaderOnBottomNav {
+      }
+      clickLibraryOnBottomNav {
+        clickFileTransferIcon {
+        }
+      }
+      clickDownloadOnBottomNav {
+        clickOnGlobeIcon {
+        }
+      }
+      openDrawer()
```

Comment:
we could move `openNavDrawer` into these `click*OnNavDrawer` methods, we could even extract a common method and it would explain those double `onBackPress` function calls better eg

```
fun inNavDrawer(navDrawerAction:()->Unit){
openNavDrawer()
navDrawerAction.invoke()
pressBack()
}
```

Reply:
Beautiful!!
I meant implementing the same functionality in test when I asked for the robots for navDrawer and bottomNav, as at that time the only way I knew of implementing it was via robots only. But this is just awesome, I didn't know about it, Wow!! Thanks!